### ISSUE TITLE
example stuck

### ISSUE THREAD

**Issue by jasmainak** [neuropycon/ephypype#28]
@annapasca I tried to run the example from the branch in your PR #22 . Now, it doesn't throw an error anymore but it gets stuck at some point. This is how far I could go:

```py
/home/mainak/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1405: UserWarning: 
This call to matplotlib.use() has no effect because the backend has already
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

  warnings.warn(_use_error_msg)
1.1.2
*** main_path -> /home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega ***
180911-10:43:31,747 nipype.workflow INFO:
	 Generated workflow graph: /home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/graph.png (graph2use=colored, simple_form=True).
180911-10:43:32,18 nipype.workflow INFO:
	 Workflow preprocessing_pipeline settings: ['check', 'execution', 'logging', 'monitoring']
180911-10:43:32,107 nipype.workflow INFO:
	 Running in parallel.
180911-10:43:32,132 nipype.workflow INFO:
	 [MultiProc] Running 0 tasks, and 1 jobs ready. Free memory (GB): 27.95/27.95, Free processors: 3/3.
180911-10:43:32,209 nipype.workflow INFO:
	 [Node] Setting-up "preprocessing_pipeline.datasource" in "/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/_sess_index_ses-0001_subject_id_sub-0003/datasource".
180911-10:43:32,249 nipype.workflow INFO:
	 [Node] Running "datasource" ("nipype.interfaces.io.DataGrabber")
180911-10:43:32,277 nipype.workflow INFO:
	 [Node] Finished "preprocessing_pipeline.datasource".
180911-10:43:34,130 nipype.workflow INFO:
	 [Job 0] Completed (preprocessing_pipeline.datasource).
180911-10:43:34,135 nipype.workflow INFO:
	 [MultiProc] Running 0 tasks, and 1 jobs ready. Free memory (GB): 27.95/27.95, Free processors: 3/3.
180911-10:43:34,255 nipype.workflow INFO:
	 [Job 1] Cached (preprocessing_pipeline.preproc_meeg.ds2fif).
180911-10:43:36,241 nipype.workflow INFO:
	 [Job 2] Cached (preprocessing_pipeline.preproc_meeg.preproc).
180911-10:43:38,244 nipype.workflow INFO:
	 [Node] Setting-up "preprocessing_pipeline.preproc_meeg.ica" in "/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/ica".
180911-10:43:38,261 nipype.workflow INFO:
	 [Node] Running "ica" ("ephypype.interfaces.mne.preproc.CompIca")
Opening raw data file /home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/preproc/sub-0003_task-rest_run-01_meg_raw_filt_dsamp.fif...
/home/mainak/Desktop/projects/github_repos/ephypype/ephypype/preproc.py:50: RuntimeWarning: This filename (/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/preproc/sub-0003_task-rest_run-01_meg_raw_filt_dsamp.fif) does not conform to MNE naming conventions. All raw files should end with raw.fif, raw_sss.fif, raw_tsss.fif, raw.fif.gz, raw_sss.fif.gz or raw_tsss.fif.gz
  raw = read_raw_fif(fif_file, preload=True)
This filename (/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/preproc/sub-0003_task-rest_run-01_meg_raw_filt_dsamp.fif) does not conform to MNE naming conventions. All raw files should end with raw.fif, raw_sss.fif, raw_tsss.fif, raw.fif.gz, raw_sss.fif.gz or raw_tsss.fif.gz
180911-10:43:40,133 nipype.workflow INFO:
	 [MultiProc] Running 1 tasks, and 0 jobs ready. Free memory (GB): 27.75/27.95, Free processors: 2/3.                     
                     Currently running:
                       * preprocessing_pipeline.preproc_meeg.ica
    Read 5 compensation matrices
    Range : 0 ... 479999 =      0.000 ...   599.999 secs
Ready.
Current compensation grade : 3
Reading 0 ... 479999  =      0.000 ...   599.999 secs...
Fitting ICA to data using 270 channels (please be patient, this may take a while)
Inferring max_pca_components from picks
    Rejecting  epoch based on MAG : [u'MLO12-4408', u'MLO13-4408', u'MLO23-4408']
Artifact detected in [294400, 296000]
Selection by explained variance: 27 components
Fitting ICA took 69.6s.
/home/mainak/Desktop/projects/github_repos/ephypype/ephypype/preproc.py:70: RuntimeWarning: This filename (/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/ica/sub-0003_task-rest_run-01_meg_raw_filt_dsamp_ica-tseries.fif) does not conform to MNE naming conventions. All raw files should end with raw.fif, raw_sss.fif, raw_tsss.fif, raw.fif.gz, raw_sss.fif.gz or raw_tsss.fif.gz
  ica_src.save(ica_ts_file)
This filename (/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/ica/sub-0003_task-rest_run-01_meg_raw_filt_dsamp_ica-tseries.fif) does not conform to MNE naming conventions. All raw files should end with raw.fif, raw_sss.fif, raw_tsss.fif, raw.fif.gz, raw_sss.fif.gz or raw_tsss.fif.gz
Writing /home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/ica/sub-0003_task-rest_run-01_meg_raw_filt_dsamp_ica-tseries.fif
Closing /home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega/preprocessing_pipeline/preproc_meeg/_sess_index_ses-0001_subject_id_sub-0003/ica/sub-0003_task-rest_run-01_meg_raw_filt_dsamp_ica-tseries.fif [done]
Using channel ECG to identify heart beats.
Setting up band-pass filter from 8 - 16 Hz
Filter length of 8192 samples (10.240 sec) selected
Number of ECG events detected : 529 (average pulse 52 / min.)
529 matching events found
No baseline correction applied
Not setting metadata
Loading data for 529 events and 801 original time points ...
0 bad epochs dropped
Reconstructing ECG signal from Magnetometers
*** EOG CHANNELS FOUND ***
Using channel HEOG and VEOG as EOG channels
... filtering ICA sources
Setting up band-pass filter from 1 - 10 Hz
Filter length of 8192 samples (10.240 sec) selected
... filtering target
Setting up band-pass filter from 1 - 10 Hz
Filter length of 8192 samples (10.240 sec) selected
... filtering ICA sources
Setting up band-pass filter from 1 - 10 Hz
Filter length of 8192 samples (10.240 sec) selected
... filtering target
Setting up band-pass filter from 1 - 10 Hz
Filter length of 8192 samples (10.240 sec) selected
Using channel HEOG and VEOG as EOG channels
EOG channel index for this subject is: [298 299]
Filtering the data to remove DC offset to help distinguish blinks from saccades
Setting up band-pass filter from 2 - 45 Hz
Filter length of 8192 samples (10.240 sec) selected
Setting up band-pass filter from 2 - 45 Hz
Filter length of 8192 samples (10.240 sec) selected
Setting up band-pass filter from 1 - 10 Hz
Filter length of 8192 samples (10.240 sec) selected
Now detecting blinks and generating corresponding events
Number of EOG events detected : 22
22 matching events found
No baseline correction applied
Not setting metadata
Loading data for 22 events and 801 original time points ...
2 bad epochs dropped
Embedding : jquery-1.10.2.min.js
Embedding : jquery-ui.min.js
Embedding : bootstrap.min.js
Embedding : jquery-ui.min.css
Embedding : bootstrap.min.css

```

**Comment by annapasca** [comment]
uh...I tried in my laptop and it works...what folder do u have in the main_path of the workflow?

**Comment by annapasca** [comment]
@davidmeunier79 @dmalt coult it be related to nipype?

**Comment by jasmainak** [comment]
This is my params file:

```py
""" Parameters file for run_preprocess_pipeline.py """

data_type = 'ds'

main_path = '/home/mainak/Desktop/projects/github_repos/BIDS-examples/sample_BIDS_omega'
data_path = main_path

subject_ids = ['sub-0003']  # 'sub-0004', 'sub-0006'
sessions = ['ses-0001']

down_sfreq = 800
l_freq = 0.1
h_freq = 150


# ------------------------ SET ICA variables -----------------------------#

is_ICA = True  # if True we apply ICA to remove ECG and EoG artifacts

# specify ECG and EoG channel names if we know them
ECG_ch_name = 'ECG'
EoG_ch_name = 'HEOG, VEOG'
variance = 0.95

reject = dict(mag=5e-12, grad=5000e-13)

# pipeline directory within the main_path dir
preproc_pipeline_name = 'preprocessing_pipeline'
```

Maybe I don't have the latest version of OMEGA dataset? But that's weird that it gets stuck because of that ...

**Comment by annapasca** [comment]
which folders do you have in the preprocessing_pipeline folder?

**Comment by annapasca** [comment]
the ds2fif and preproc Node work well, the problem is in the ICA node that computes all (ICA, ecg and eog epochs). 
It seems to me a problem related to the generation of report. I'll look in

**Comment by annapasca** [comment]
I'm testing all in the mne enviroment. Are you using this env?

**Comment by dmalt** [comment]
@annapasca, Sorry, I'm a bit short on time till Friday. I can look on weekend if the issue isn't resolved.

**Comment by annapasca** [comment]
@dmalt don't worry!!! I tried to fix it in #22

**Comment by annapasca** [comment]
> @annapasca we can close this issue if you update the README to say that `ephypype` works only on `Python 3`

Done, but I have some problem with pull request... I tried twice (#61 #62) but there is something weird today
